src/main/java/org/tron/keystore/WalletFile.java

```diff
@@ -0,0 +1,482 @@
+        return false;
+      }
+      if (getCipherparams() != null
+          ? !getCipherparams().equals(that.getCipherparams())
+          : that.getCipherparams() != null) {
+        return false;
+      }
+      if (getKdf() != null
+          ? !getKdf().equals(that.getKdf())
+          : that.getKdf() != null) {
+        return false;
+      }
+      if (getKdfparams() != null
+          ? !getKdfparams().equals(that.getKdfparams())
+          : that.getKdfparams() != null) {
+        return false;
+      }
+      return getMac() != null
+          ? getMac().equals(that.getMac()) : that.getMac() == null;
+    }
+
+    @Override
+    public int hashCode() {
+      int result = getCipher() != null ? getCipher().hashCode() : 0;
+      result = 31 * result + (getCiphertext() != null ? getCiphertext().hashCode() : 0);
+      result = 31 * result + (getCipherparams() != null ? getCipherparams().hashCode() : 0);
+      result = 31 * result + (getKdf() != null ? getKdf().hashCode() : 0);
+      result = 31 * result + (getKdfparams() != null ? getKdfparams().hashCode() : 0);
+      result = 31 * result + (getMac() != null ? getMac().hashCode() : 0);
+      return result;
+    }
+
+  }
+
+  public static class CipherParams {
+
+    private String iv;
+
+    public CipherParams() {
+    }
+
+    public String getIv() {
+      return iv;
+    }
+
+    public void setIv(String iv) {
+      this.iv = iv;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null) {
+        return false;
+      }
+      if (o.getClass() != CipherParams.class) {
+        return false;
+      }
+
+      CipherParams that = (CipherParams) o;
+
+      return getIv() != null
+          ? getIv().equals(that.getIv()) : that.getIv() == null;
+    }
+
+    @Override
+    public int hashCode() {
+      int result = getIv() != null ? getIv().hashCode() : 0;
+      return result;
+    }
+
+  }
+
+  interface KdfParams {
+
+    int getDklen();
+
+    String getSalt();
+  }
+
+  public static class Aes128CtrKdfParams implements KdfParams {
+
+    private int dklen;
+    private int c;
+    private String prf;
+    private String salt;
+
+    public Aes128CtrKdfParams() {
+    }
+
+    public int getDklen() {
+      return dklen;
+    }
+
+    public void setDklen(int dklen) {
+      this.dklen = dklen;
+    }
+
+    public int getC() {
+      return c;
+    }
+
+    public void setC(int c) {
+      this.c = c;
+    }
+
+    public String getPrf() {
+      return prf;
+    }
+
+    public void setPrf(String prf) {
+      this.prf = prf;
+    }
+
+    public String getSalt() {
+      return salt;
+    }
+
+    public void setSalt(String salt) {
+      this.salt = salt;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null) {
+        return false;
+      }
+      if (o.getClass() != Aes128CtrKdfParams.class) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Compare to "this.getClass()" instead. ([squid:S2162](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2162))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=938&issues=AWPK23iWkDisRwWBfuxJ&open=AWPK23iWkDisRwWBfuxJ)

[ISSUE_KEY]: AWPK23iWkDisRwWBfuxJ